### PR TITLE
Add compute kernel config input to batch norm

### LINKED ISSUE #24167: Add compute_kernel_config argument to batch_norm
### Is your feature request related to a problem? Please describe

Batch norm compute kernel is not configurable like other ops

### Describe the solution you'd like

Add and expose the compute_kernel_config parameter to batch_norm as is done for other ops

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rmillerTT**
### Ticket
#24167

### Problem description
Compute kernel config is not an input to batch norm like other ops. It should be provided to the user to control compute accuracy

### What's changed
Plumbed through the compute config and exposed it to Python

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes https://github.com/tenstorrent/tt-metal/actions/runs/16324224758
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16354552065
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16324236057
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16324242592
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [x] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-quick-trigger.yaml) CI passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16324258193
- [ ] [TG demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [x] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main) https://github.com/tenstorrent/tt-metal/actions/runs/16324272889
- [x] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release) https://github.com/tenstorrent/tt-metal/actions/runs/16324278373

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/fused/test_batch_norm.py [hunk 1]
```diff
@@ -422,3 +422,103 @@ def test_batch_norm_output_Default(input_shapes, device):
...
+
+    def compute_pccs_for_tensors(torch_tensors, tt_tensors):
+        pccs = []
+        for torch_tensor, tt_tensor in zip(torch_tensors, tt_tensors):
+            _, pcc = comp_pcc(torch_tensor, tt_tensor)
+            pccs.append(pcc)
+        return pccs
+
+    # Execute low-accuracy groupnorm
+    config_low = ttnn.WormholeComputeKernelConfig(
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/fused/test_batch_norm.py]: Please use ttnn.init_device_compute_kernel_config instead to avoid naming device specific code.

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const std::optional<DeviceComputeKernelConfig>& compute_kernel_config, const Tensor& input) {
+    TT_FATAL(
+        input.storage_type() == StorageType::DEVICE,
+        "Invalid input tensor storage type: Input tensor must be on device. (storage type={})",
+        input.storage_type());
+
+    const auto arch = input.device()->arch();
+    const auto input_data_format = datatype_to_dataformat_converter(input.dtype());
+    const auto default_math_fidelity = MathFidelity::HiFi4;
+    const auto default_approx_mode = true;
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp]: I think this should be false. Is there a reason this is true?

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    const auto default_fp32_acc = input_data_format == tt::DataFormat::UInt32 ||
+                                  input_data_format == tt::DataFormat::Int32 ||
+                                  input_data_format == tt::DataFormat::Float32;
+    const auto default_l1_acc = false;
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp]: Could you please check if changing this to true has any impact?

File: tests/ttnn/unit_tests/operations/fused/test_batch_norm.py (hunk 1 above)
Comment (rmillerTT) [reply tests/ttnn/unit_tests/operations/fused/test_batch_norm.py]: Oh, didn't know that init was wrapped. Will do

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp (hunk 2 above)
Comment (rmillerTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp]: My mistake, it's fixed.

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp (hunk 3 above)
Comment (rmillerTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_utils.cpp]: Changed to true and batch norm unit tests still pass, will run pipelines now

**Review by bbradelTT — APPROVED** [review]

**Comment by rmillerTT** [comment]
SC demo tests had 2 failures: 
- 1 is a CI issue
- The other was a model slowdown that is also in main
A re-run was triggered but was queued for basically the entire day.

TG demo tests had one perf failure that is also in main.

Since all other tests passed, this looks good to merge.
